### PR TITLE
tighten permissions on configuration data files

### PR DESCRIPTION
**Pull request by sallyom**
also, small fix to the apiserver-audit-log path

**Which issue(s) this PR addresses**:

Closes [USHIFT-84](https://issues.redhat.com/browse/USHIFT-84)

### CODE REVIEW

**Review by ggiguash — CHANGES_REQUESTED** [review]
Since we're specifying a mask/mode for directory creation, it must be at least 0700 to have x bit set for the owner. Otherwise, cd would not work into that directory

**Review by ggiguash — APPROVED** [review]

File: pkg/controllers/openshift-apiserver.go [hunk 1]
```diff
@@ -196,7 +196,7 @@ apiServerArguments:
 	if cfg.AuditLogDir != "" {
 		data = append(data, `
 auditConfig:
-  auditFilePath: `+cfg.AuditLogDir+`openshift-apiserver-audit.log
+  auditFilePath: "`+cfg.AuditLogDir+`/openshift-apiserver-audit.log"
```
Comment (dhellmann) [inline pkg/controllers/openshift-apiserver.go:199]: Will the need for that slash depend on whether the AuditLogDir variable ends in one? Should we use filepath.Join()?

**Review by dhellmann — COMMENTED** [review]
I had one question that can be addressed in another PR if needed.

/lgtm

**Comment by ggiguash** [comment]
USHIFT-84

**Pull request merged**